src/main.cpp

```diff
@@ -244,7 +244,9 @@ class ArgFactory {
   private:
     char *AllocateString(std::string string) {
         int len = string.length();
-        char *ptr = new char[len + 1];
+        // We use malloc here because of strdup in lAddSingleArg
```

Comment:
I'm not sure how is this related. Could you elaborate?

Reply:
Ah, I see. You want to deallocate both of them in the same manner and it ha to be `free()`.

I think a better solution would be to use only C++ primitives to work with strings, but this that would require more changes. So this is ok.

Reply:
Yes, exactly.
`argv` vector consisted of pointers to C string of two origins:
- real arguments of program from `Argv` of `main`
- heap allocated strings.
They can be mixed in order inside `argv` so I didn't find a easy way to distinguish them in order to deallocate the last ones.

I have ended up with the current fix that `strdup` the first ones and allocate the last ones with `malloc` to deallocate all of them with `free`.